### PR TITLE
Enable lost pixel for designer app

### PR DESCRIPTION
**Pull request by TrySound**
We have a lot of features in designer to make snapshot of.

## Code Review

- [ ] hi @kof, I need you to do
  - detailed review (read every line)

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

**Review by kof — APPROVED** [review]

**Comment by TrySound** [comment]
Hi @d-ivashchuk. I'm trying to setup lost-pixel for another project and I'm not sure it works. Could you check if I'm doing something wrong?

**Comment by d-ivashchuk** [comment]
hey @TrySound! couple of questions :D is this monorepo lost pixel usage now? would it be used across packages?

@kof we did agree that migrating to bleeding edge lost-pixel makes sense, we could do monorepo support & migration in this PR already. 

Final question - does the team relies on `git rebase`? we are having troubles with this flow as it messes up git history for the repo that we rely on in `lost-pixel-v1`. We are working on the fix, but currently if you are rebasing I would not recommend to switch to next version. 

Legacy lost-pixel that you are using now `gitbased.app` also has monorepo support so no issues, we can make it work as well in the current setup.

Thanks for the answers in advance!

**Comment by kof** [comment]
@d-ivashchuk for some reason I thought we already switched, I think we could avoid rebasing, I almost never rebase personally, but we had a lengthy discussion around it with no particular outcome, because there are tradeoffs for both

**Comment by kof** [comment]
Rebasing is only a problem in wwhich sense? it won't pickup the changes?

**Comment by d-ivashchuk** [comment]
@kof we just agreed that we want to switch :D rebasing will destroy your baseline approvals, so if you will want to create new PR -> approve -> merge normally, it will be ok. nothing that can't be remedied with this. It's different story for teams that **_just_** rebase, we have troubles with that now. but should be working as well, should not be a blocker. hopefully @chriskalmar will find the fix in the nearest future.

**Comment by TrySound** [comment]
We do squash into single commit before merging and rebase PRs from main.

**Comment by d-ivashchuk** [comment]
> We do squash into single commit before merging and rebase PRs from main.

this could be a trouble for now :/ would still be interesting to see, but I don't want to break your current lost-pixel flows just to see that it does not work. give us some time to figure out the problem with squashes + rebases and then we will safely migrate you to the new version of the platform :D 

for now if you can't avoid rebasing - let's stick to file based lost-pixel. I will send you some examples of how the monorepo mode works! but first  I did not see the confirmation for this:

> is this monorepo lost pixel usage now? would it be used across packages?

Did I get this right?

**Comment by kof** [comment]
@d-ivashchuk yes, its always been a monorepo, it was just used before in one package and now we added app support and later other packages as well

**Comment by d-ivashchuk** [comment]
perfect! I can't promise it today because relatives came and I need to entertain them for a while, but over the weekend I will help you set the project up for `gitbased` monorepo, and then as soon as we figure rebases we will migrate you to next :D

**Comment by TrySound** [comment]
We would not have the issue with current setup if we had single storybook setup for whole monorepo.

File: .github/workflows/vis-reg-tests.yml [hunk 1]
```diff
@@ -10,12 +10,24 @@ jobs:
...
           S3_ACCESS_KEY: ${{ secrets.S3_ACCESS_KEY }}
           S3_SECRET_KEY: ${{ secrets.S3_SECRET_KEY }}
           LOST_PIXEL_CONFIG_DIR: packages/design-system
+
+      - name: Lost Pixel for Designer
+        uses: lost-pixel/lost-pixel@v2.23.0
+        env:
+          S3_ACCESS_KEY: ${{ secrets.S3_ACCESS_KEY }}
+          S3_SECRET_KEY: ${{ secrets.S3_SECRET_KEY }}
+          LOST_PIXEL_CONFIG_DIR: apps/designer
```
Comment (d-ivashchuk) [inline .github/workflows/vis-reg-tests.yml:33]:
```suggestion
      - name: Lost Pixel for Design System
        uses: lost-pixel/lost-pixel@v2.23.0
        env:
          S3_ACCESS_KEY: ${{ secrets.S3_ACCESS_KEY }}
          S3_SECRET_KEY: ${{ secrets.S3_SECRET_KEY }}
          LOST_PIXEL_CONFIG_DIR: packages/design-system

      - name: Lost Pixel for Designer
        uses: lost-pixel/lost-pixel@v2.23.0
        env:
          S3_ACCESS_KEY: ${{ secrets.S3_ACCESS_KEY }}
          S3_SECRET_KEY: ${{ secrets.S3_SECRET_KEY }}
          LOST_PIXEL_CONFIG_DIR: apps/designer
          
      - name: Lost Pixel
        uses: lost-pixel/lost-pixel@v2.24.1
        with:
          FINALIZE: true
```

Comment (d-ivashchuk) [reply .github/workflows/vis-reg-tests.yml:33]: Happy new year guys!

Please tell me when you are ready to work on this & try this out. Then I'd need to flick the monorepo support for you manually & you will need to accept these changes :D

Comment (TrySound) [reply .github/workflows/vis-reg-tests.yml]: Happy new year! Let's try this

File: .github/workflows/vis-reg-tests.yml [hunk 2]
```diff
@@ -10,12 +10,31 @@ jobs:
...
+          
+      - name: Lost Pixel
+        uses: lost-pixel/lost-pixel@v2.24.1
+        env:
+          LOST_PIXEL_CONFIG_DIR: apps/designer
+        with:
+          FINALIZE: true
```
Comment (d-ivashchuk) [inline .github/workflows/vis-reg-tests.yml:40]:
```suggestion
      - name: Lost Pixel
        uses: lost-pixel/lost-pixel@v2.24.1
        env:
          LOST_PIXEL_CONFIG_DIR: apps/designer
          S3_ACCESS_KEY: ${{ secrets.S3_ACCESS_KEY }}
          S3_SECRET_KEY: ${{ secrets.S3_SECRET_KEY }}
        with:
          FINALIZE: true
```

**Review by d-ivashchuk — COMMENTED** [review]
we need to add this(still an after effect of us using legacy version, once we switch to the latest lost-pixel will be far less boilerplate here)

File: packages/design-system/lostpixel.config.ts [hunk 3]
```diff
@@ -1,12 +1,11 @@
-import type { CustomProjectConfig } from "lost-pixel";
 import * as process from "process";
 
-export const config: CustomProjectConfig = {
+export const config = {
   storybookShots: {
     storybookUrl: "packages/design-system/storybook-static",
   },
   lostPixelProjectId: "cl8a8xmx714904901m9oce4vhqc",
```
Comment (d-ivashchuk) [inline packages/design-system/lostpixel.config.ts:7]:
```suggestion
  lostPixelProjectId: "clckoyuix110903101s75d4s71fc",
```

**Comment by kof** [comment]
Seems like we are stuck here? the question is what blocks us, we have all the help we can get from Lostpixel authors

**Comment by kof** [comment]
closing this one since @andrasbacsai has worked on it separately

**Pull request closed without merging**